src/ipc/ipc4/handler.c

```diff
@@ -352,15 +347,23 @@ static int ipc4_bind_module(union ipc4_message_header *ipc4)
 					(uint32_t)bu.header.r.instance_id, (uint32_t)bu.data.r.dst_module_id,
 					(uint32_t)bu.data.r.dst_instance_id);
 
-	comp[0] = IPC4_COMP_ID(bu.header.r.module_id, bu.header.r.instance_id);
-	comp[1] = IPC4_COMP_ID(bu.data.r.dst_module_id, bu.data.r.dst_instance_id);
 
-	return ipc_comp_connect(ipc, comp);
+	return ipc_comp_connect(ipc, (ipc_pipe_comp_connect *)&bu);
 }
 
 static int ipc4_unbind_module(union ipc4_message_header *ipc4)
 {
-	return 0;
+	struct ipc4_module_bind_unbind bu;
+	struct ipc *ipc = ipc_get();
+
+	bu.header.dat = *ipc_to_hdr(ipc4);
```

Comment:
Cant we just memcpy this from ipc to bu ?

Reply:
sure